src/main/java/org/opentripplanner/transit/model/network/TripPatternBuilder.java

```diff
@@ -86,10 +93,28 @@ public TripPatternBuilder withStopPattern(StopPattern stopPattern) {
   }
 
   public TripPatternBuilder withScheduledTimeTable(Timetable scheduledTimetable) {
+    if (scheduledTimetableBuilder != null) {
+      throw new IllegalStateException(
+        "Cannot set scheduled Timetable after scheduled Timetable builder is created"
+      );
+    }
     this.scheduledTimetable = scheduledTimetable;
     return this;
   }
 
+  public TripPatternBuilder withScheduledTimeTableBuilder(
+    UnaryOperator<TimetableBuilder> producer
+  ) {
+    // create a builder for the scheduled timetable only if it needs to be modified.
+    // otherwise reuse the existing timetable
+    if (scheduledTimetableBuilder == null) {
+      scheduledTimetableBuilder = scheduledTimetable.copyOf();
+      scheduledTimetable = null;
+    }
+    producer.apply(scheduledTimetableBuilder);
+    return this;
+  }
+
```

Comment:
An alternative to holding on to the scheduledTimetableBuilder is to hold on to the scheduledTimetable instead. Depending on what we ned we could hold onto the modified object and/or original.

Reply:
This is not possible due to the bidirectional relation between TripPattern and Timetable: if you built the scheduled Timetable in TripPatternBuilder, you would need to set its reference to TripPattern, which is being built.